frontend/src/metabase-lib/expressions/resolver.js

```diff
@@ -73,8 +73,14 @@ const isCompatible = (a, b) => {
  * @param {import("./pratt").Expr} expression
  * @param {string} type
  * @param {?(kind: string, name: string, node: import("./pratt").Node) => void} fn
+ * @param {Database | null} database
  */
-export function resolve(expression, type = "expression", fn = undefined) {
+export function resolve(
+  expression,
+  type = "expression",
+  fn = undefined,
+  database,
```

Comment:
nit: optional arguments should go last

Maybe in this case we'd be better off with a param object instead of N arguments.